File: app-react/src/pages/flutter/routers/buttonRouter.js
﻿/* ====================================== 模块子路由配置  ====================================== */
import React from 'react'
import { Switch, Route } from 'react-router-dom'
// ===================================================================== 异步加载
import Import from '@common/bundle'
// ===================================================================== 二级路由
export default class ButtonRouter extends React.Component{
	render(){
		const { match } = this.props
		return (
				<Switch>
					<Route path={`${match.url}/MaterialButton`} 		component={ Import('flutter/widget/button/MaterialButton') } exact />
					<Route path={`${match.url}/RaisedButton`} 			component={ Import('flutter/widget/button/RaisedButton') } exact />
					<Route path={`${match.url}/FlatButton`} 			component={ Import('flutter/widget/button/FlatButton') } exact />
					<Route path={`${match.url}/OutlineButton`} 			component={ Import('flutter/widget/button/OutlineButton') } exact />
					<Route path={`${match.url}/FloatingActionButton`} 	component={ Import('flutter/widget/button/FloatingActionButton') } exact />
					<Route path={`${match.url}/DropdownButton`} 		component={ Import('flutter/widget/button/DropdownButton') } exact />
					<Route path={`${match.url}/IconButton`} 			component={ Import('flutter/widget/button/IconButton') } exact />
					<Route path={`${match.url}/ButtonBar`} 				component={ Import('flutter/widget/button/ButtonBar') } exact />
				</Switch>
		)
	}
}

File: app-flutter/lib/router/buttonRouter.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 路由
import 'package:flutter_web_study.app/pages/flutter/widget/button/MaterialButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/RaisedButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/OutlineButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/FlatButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/FloatingActionButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/DropdownButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/IconButton.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/button/ButtonBar.dart';
// ======================================================================================================== 路由类
class ButtonRouter{
	final Map<String,WidgetBuilder> router = <String,WidgetBuilder>{
		'/MaterialButtonPage': (BuildContext context) => MaterialButtonPage(title:'MaterialButton( )'),
		'/RaisedButtonPage': (BuildContext context) => RaisedButtonPage(title:'RaisedButton( )'),
		'/OutlineButtonPage': (BuildContext context) => OutlineButtonPage(title:'OutlineButton( )'),
		'/FlatButtonPage': (BuildContext context) => FlatButtonPage(title:'FlatButton( )'),
		'/FloatingActionButtonPage': (BuildContext context) => FloatingActionButtonPage(title:'FloatingActionButton( )'),
		'/DropdownButtonPage': (BuildContext context) => DropdownButtonPage(title:'DropdownButton( )'),
		'/IconButtonPage': (BuildContext context) => IconButtonPage(title:'IconButton( )'),
		'/ButtonBarPage': (BuildContext context) => ButtonBarPage(title:'ButtonBar( )'),
	};
}